.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,167 @@
+name: "DotNet SDK Upgrade"
+
+on:
+  schedule:
+    - cron: '0 0 * * 1'
+  workflow_dispatch:
+
+env: 
+  MAJOR_MINOR_VERSION: "6.0"
+
+jobs:
+  dotnet-update:
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{ steps.create_branch.outputs.BRANCH_NAME }}
+      LATEST_PATCH_VERSION: ${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}
+      SHOULD_UPDATE: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Check patch version
+        id: fetch_latest_patch
+        shell: bash
+        run: |
+          latest_patch_version=$(curl -sb -H "Accept: application/json" "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${{ env.MAJOR_MINOR_VERSION }}/latest.version")
+          current_patch_version=$(jq .sdk.version ./src/global.json | xargs)
+
+          exit_code=0
+          function version { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+          
+          [ $(version $current_patch_version) -ge $(version $latest_patch_version) ] && exit_code=1
+
+          echo "DOTNET_CURRENT_VERSION=${current_patch_version}" >> $GITHUB_OUTPUT
+          echo "DOTNET_LATEST_VERSION=${latest_patch_version}" >> $GITHUB_OUTPUT
+          echo "SHOULD_UPDATE=${exit_code}" >> $GITHUB_OUTPUT
+      - name: Create annotation for build error
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 1 }}
+        run: echo "::warning ::Latest DotNet SDK patch is ${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}, and we are on ${{ steps.fetch_latest_patch.outputs.DOTNET_CURRENT_VERSION }}. No need to update"
+      - name: Update patch version
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 0 }}
+        shell: bash
+        run: |
+          patch_version="${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}"
+          current_version="${{ steps.fetch_latest_patch.outputs.DOTNET_CURRENT_VERSION }}"
+
+          # Update globals
+          echo $(cat ./src/global.json | jq --arg patch_version "$patch_version" '.sdk.version = $patch_version') > ./src/global.json
+
+          # Update devcontainer
+          echo $(cat ./.devcontainer/devcontainer.json | jq --arg patch_version "$patch_version" '.features."ghcr.io/devcontainers/features/dotnet".version = $patch_version') > ./.devcontainer/devcontainer.json
+
+          # Update dev.sh
+          sed -i "s/DOTNETSDK_VERSION=\"$current_version\"/DOTNETSDK_VERSION=\"$patch_version\"/g" ./src/dev.sh
+      - name: GIT commit and push all changed files
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 0 }}
+        id: create_branch
+        env: 
```

Comment:
do we need these ENV if we are not reuse them? maybe just inline the string in the `run` script

Reply:
Yeah I put them there just as an easier way to access the variables instead of checking the code itself, Will move it down though since its quite small